stellargraph/mapper/sequences.py

```diff
@@ -527,3 +528,40 @@ def __len__(self):
 
     def __getitem__(self, index):
         return self.inputs, self.targets
+
+
+class CorruptedNodeSequence(Sequence):
+    """
+    Keras compatible data generator that wraps a FullBatchSequence ot SparseFullBatchSequence and provides corrupted
+    data for training Deep Graph Infomax.
+
+    Args:
+        base_generator: the uncorrupted Sequence object.
+    """
+
+    def __init__(self, base_generator):
+
+        if not isinstance(
+            base_generator, (FullBatchSequence, SparseFullBatchSequence,),
+        ):
+            raise TypeError(
+                f"base_generator: expected FullBatchSequence or SparseFullBatchSequence, "
+                f"found {type(base_generator).__name__}"
+            )
+
+        self.base_generator = base_generator
+        self.targets = np.zeros((1, len(base_generator.target_indices), 2))
+        self.targets[0, :, 0] = 1.0
+
+    def __len__(self):
+        return self.base_generator.__len__()
```

Comment:
```suggestion
        return len(self.base_generator)
```

(Usually, one should never call a dunder method like `__len__` or `__getitem__` if there's an alternative: the alternatives are cleaner.)